flake.nix

```diff
@@ -0,0 +1,48 @@
+{
+  description = "Systems language with linear types and capability-based security.";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
+  inputs.utils.url = "github:numtide/flake-utils";
+
+  outputs = { self, nixpkgs, utils }: 
+    utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        buildInputs = with pkgs; [
+          # General
+          gmp
+          python311
+
+          # Tooling
+          ocamlPackages.ocaml
+          ocamlPackages.dune_3
+          ocamlPackages.findlib
+          ocamlPackages.odoc
+
+          # OCaml libraries
+          ocamlPackages.yojson
+          ocamlPackages.ppx_deriving
+          ocamlPackages.ounit2
+          ocamlPackages.menhir
+          ocamlPackages.sexplib
+          ocamlPackages.ppx_sexp_conv
+          ocamlPackages.zarith
+        ];
+
+      in {
+        packages.default = pkgs.stdenv.mkDerivation {
+          pname = "austral";
+          version = "0.2.0";
+          src = ./.;
+          inherit buildInputs;
+          installPhase = ''
+            mkdir -p $out/bin
+            install -m 755 austral $out/bin/austral
```

Comment:
The default mkDerivation installPhase calls `make install`, so we could theoretically eliminate this duplication with the makeFile like so:

### Makefile:
```Makefile
PREFIX ?= /usr/local
# ...
# install -D creates all required directories - `bin` in this case
install: $(BIN)
	install -D -m 755 austral $(PREFIX)/bin/austral
```
### flake.nix:
```nix
pkgs.stdenv.mkDerivation {
  # ...
  installFlags = [ "PREFIX=$(out)" ];
  # remove installPhase
}
```

Only problem is, I'm not sure what to do about `make uninstall`.

Reply:
I think it's fine to go ahead with this, since the only time PREFIX would be $(out) is during a `nix build`, and there's no reason that would ever be running `make uninstall`.